### ISSUE TITLE
gfm-view-mode set fontification default

### ISSUE THREAD

**Issue by theothornhill** [jrblevin/markdown-mode#451]
## Expected Behavior
Improvement: can `gfm-view-mode` fontify code blocks natively by default?

## Actual Behavior
It isn't default, but is available with `(setq markdown-fontify-code-blocks-natively t)`.

The reason I ask for this, is that [eglot](https://github.com/joaotavora/eglot) is using `gfm-view-mode` to render help docs. Right now it doesn't fontify by default. Since I guess eglot won't try to configure other packages too deeply, then users have to set it manually. See [discussion](https://github.com/joaotavora/eglot/pull/408) here.

Is there a reason this option is not enabled? I see you set some local variables already when deriving this mode.

**Comment by syohex** [comment]
I'm not sure however that fontification function is too heavy function. It calls each major-mode function(like `cc-mode`, `python-mode` etc) for fontification code block and those hooks are also called. If user set heavy hook function like lsp-mode, a mode which launches server program etc, markdown-mode is blocked until they finished. I suppose this makes annoyed for some people.

So I suppose it is better that its value is `nil` by default rather then `t`

**Comment by theothornhill** [comment]
> I'm not sure however that fontification function is too heavy function.

I understand, however, I'm not suggesting to add it natively in normal `markdown-mode` or `gfm-mode`, but only in `gfm-view-mode`

I'm assuming that when users want to view the markdown in `gfm`, it is likely they want to see the syntax colors as well.

> (...) a mode which launches server program etc, markdown-mode is blocked until they finished. 

I'm not sure I understand what you mean here. What is done in `eglot` is only using `markdown-mode` after response has arrived.

I'm sure I have missed something - this is however my suggestion:

```elisp
(define-derived-mode gfm-view-mode gfm-mode "GFM-View"
  "Major mode for viewing GitHub Flavored Markdown content."
  (setq-local markdown-hide-markup markdown-hide-markup-in-view-modes
              markdown-fontify-code-blocks-natively t)
  (read-only-mode 1))

```

**Comment by syohex** [comment]
@theothornhill I have merged #453. Please check latest version.
